### DISCUSSION TITLE
Train different models in parallel

### DISCUSSION THREAD

**Discussion by aliutkus** [Lightning-AI/pytorch-lightning#12548] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello, thanks for the awesome work

I would like to train several models in parallel, that would ideally share the same dataloader, but have non shared parameters, and each one its loss.

Could be implemented like passing a list to a trainer somehow.

Giving a bit of context:
* Having just unrelated processes run in parallel would _not_ be an option, because I additionally want the different models to communicate during their training, e.g. through some multiprocessing.Queue object
* Please notice I want to have _several models_ trained (possibly one one single gpu), and not a single model trained on multiple gpus, which I understand is easily done through pl

Is there a way to do that ?

thanks
